### PR TITLE
build debian package for armhf ubuntu-xenial

### PR DESCRIPTION
**Pull request by flixr**
**\- What I did**
Add a Dockerfile to build armhf deb package for Ubuntu Xenial (16.04)

**\- How to verify it**
Build deb, install, enjoy...

**\- Description for the changelog**
Build armhf debian package for Ubuntu 16.04

Signed-off-by: Felix Ruess felix.ruess@gmail.com

### CODE REVIEW

**Comment by tophj-ibm** [comment]
tested this on my rpi and everything made / installed correctly. 👍

**Comment by thaJeztah** [comment]
ping @tianon @tiborvass @andrewhsu PTAL

File: contrib/builder/deb/armhf/ubuntu-xenial/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM armhf/ubuntu:xenial
+
+RUN apt-get update && apt-get install -y apparmor bash-completion btrfs-tools build-essential curl ca-certificates debhelper dh-apparmor dh-systemd git libapparmor-dev libdevmapper-dev libltdl-dev libseccomp-dev libsqlite3-dev pkg-config libsystemd-dev --no-install-recommends && rm -rf /var/lib/apt/lists/*
+
+ENV GO_VERSION 1.7.1
```
Comment (StefanScherer) [inline contrib/builder/deb/armhf/ubuntu-xenial/Dockerfile]: This should be updated to GO_VERSION 1.7.3

File: contrib/builder/deb/armhf/ubuntu-xenial/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
+FROM armhf/ubuntu:xenial
+
+RUN apt-get update && apt-get install -y apparmor bash-completion btrfs-tools build-essential curl ca-certificates debhelper dh-apparmor dh-systemd git libapparmor-dev libdevmapper-dev libltdl-dev libseccomp-dev libsqlite3-dev pkg-config libsystemd-dev --no-install-recommends && rm -rf /var/lib/apt/lists/*
+
+ENV GO_VERSION 1.7.3
+RUN curl -fSL "https://storage.googleapis.com/golang/go${GO_VERSION}.linux-armv6l.tar.gz" | tar xzC /usr/local
```
Comment (justincormack) [inline contrib/builder/deb/armhf/ubuntu-xenial/Dockerfile]: This should be from golang.org see changes in the other ones recently

**Review by justincormack — CHANGES_REQUESTED** [review]

File: contrib/builder/deb/armhf/ubuntu-xenial/Dockerfile (hunk 2 above)
Comment (thaJeztah) [inline contrib/builder/deb/armhf/ubuntu-xenial/Dockerfile]: We should start working on an update/create a generate.sh for armhf, like we have for amd64; https://github.com/docker/docker/blob/master/contrib/builder/deb/amd64/build.sh

Comment (andrewhsu) [inline contrib/builder/deb/armhf/ubuntu-xenial/Dockerfile]: fyi justin is referring to #28068

**Comment by andrewhsu** [comment]
I've confirmed building an image from this `Dockerfile` works on `armv7l`.

File: contrib/builder/deb/armhf/ubuntu-xenial/Dockerfile (hunk 2 above)
Comment (flixr) [inline contrib/builder/deb/armhf/ubuntu-xenial/Dockerfile]: Thx, fixed.
And yes, a script would be the way to go... don't have the time for that right now though.

Comment (andrewhsu) [inline contrib/builder/deb/armhf/ubuntu-xenial/Dockerfile]: LGTM

File: contrib/builder/deb/armhf/ubuntu-xenial/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
+FROM armhf/ubuntu:xenial
+
+RUN apt-get update && apt-get install -y apparmor bash-completion btrfs-tools build-essential curl ca-certificates debhelper dh-apparmor dh-systemd git libapparmor-dev libdevmapper-dev libltdl-dev libseccomp-dev libsqlite3-dev pkg-config libsystemd-dev --no-install-recommends && rm -rf /var/lib/apt/lists/*
```
Comment (andrewhsu) [inline contrib/builder/deb/armhf/ubuntu-xenial/Dockerfile]: don't forget to keep it alphabetical :)

Comment (flixr) [inline contrib/builder/deb/armhf/ubuntu-xenial/Dockerfile]: done

**Comment by flixr** [comment]
@justincormack done

**Comment by andrewhsu** [comment]
LGTM

**Review by justincormack — APPROVED** [review]

**Review by thaJeztah — APPROVED** [review]
LGTM, thank you!

**Pull request merged**